core-bundle/src/Fragment/FragmentHandler.php

```diff
@@ -53,6 +54,10 @@ public function render($uri, string $renderer = 'inline', array $options = []):
             throw new UnknownFragmentException(sprintf('Invalid fragment identifier "%s"', $uri->controller));
         }
 
+        if ($uri->isBackendScope()) {
```

Comment:
I know that it never makes sense to have an ESI fragment in the back end because you're always logged in = there's always a cookie = there's never a cache. But I still don't think this is the right way to fix the undo problem. We might still keep this of course because as I said, I don't see any valid use case where separate fragments (not even hincluce, really) would make sense in the back end. 
Fixing the undo preview problem can imho only be fixed by rendering the HTML **before it's deleted** and store that along the undo record. Because there are so many things that might not work anymore: relations, API calls, etc. pp. //cc @contao/developers

Reply:
See https://github.com/contao/contao/pull/3994